lib/collection/src/collection/mod.rs

```diff
@@ -563,6 +563,19 @@ impl Collection {
                 continue;
             }
 
+            // Select shard transfer method
+            // If all peers are 1.8+, we try WAL delta transfer, otherwise use configured default
+            let all_support_wal_delta = self
+                .channel_service
+                .all_peers_at_version(Version::new(1, 8, 0));
+            let default_shard_transfer_method = if all_support_wal_delta {
+                ShardTransferMethod::WalDelta
+            } else {
+                self.shared_storage_config
+                    .default_shard_transfer_method
+                    .unwrap_or_default()
+            };
```

Comment:
As I've mentioned on #3798, we might want to rework this *somehow* to allow users to forcefully disable WAL diff transfers, if wanted/needed.

I think the best way would be to introduce `wal_delta_fallback_method` (or whatever name you prefer), and change this check to something like:

```suggestion
            // Select shard transfer method
            // If all peers are 1.8+, we try WAL delta transfer, otherwise use configured default
            let shard_transfer_method = self.shared_storage_config
                .default_shard_transfer_method
                .unwrap_or_default();
            
            let shard_transfer_method = match shard_transfer_method {
                ShardTransferMethod::WalDelta => {
                    let all_support_wal_delta = self
                        .channel_service
                        .all_peers_at_version(Version::new(1, 8, 0));

                    if all_support_wal_delta {
                        ShardTransferMethod::WalDelta
                    } else {
                        self.shared_storage_config
                            .wal_delta_fallback_method
                            .unwrap_or_default()
                    }
                }

                method => method,
            };
```

Reply:
What would be a good reason for users to disable this?

Reply:
What is we missed some testing scenario, and it corrupts data? And now whoever upgraded to 1.9 is stuck, with no way to disable WAL diff transfer.

Reply:
I think the only right solution would be to release a patch version right away.

Reply:
That would definitely take more time than for a customer to change config parameter and restart a node. Anyway, take it to @generall, that's his idea.

Reply:
I've changed this now.

It always uses the user configured method here. If it is `null` (default), we make a smart selection between WAL delta (if supported) and streaming records.